### PR TITLE
cleanup(libsinsp): add `ppm_sc_modifies_state` and `ppm_sc_repair_state` flags to sinsp-example

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

As the major `ppm sc` API refactor has completed extend `sinsp-example` for demonstration and e2e testing purposes. 

Select sc codes for active tracing in the kernel:
- Include each ppm sc code from filter expression AST.
- Demonstrate ppm sc API usage for both  `sinsp_state_sc_set` and `sinsp_repair_state_sc_set` options.

CC @FedeDP @Andreagit97 @jasondellaluce 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
cleanup(libsinsp): add `ppm_sc_modifies_state` and `ppm_sc_repair_state` flags to sinsp-example
```

### CODE REVIEW

**Review by FedeDP — APPROVED** [review]
/approve

File: userspace/libsinsp/examples/test.cpp [hunk 1]
```diff
@@ -318,12 +353,15 @@ int main(int argc, char** argv)
...
 
-	auto event_names = extract_filter_events(inspector);
-	if(!event_names.empty())
+	auto events_sc_codes = extract_filter_sc_codes(inspector);
+	if(!events_sc_codes.empty())
 	{
-		printf("-- Filter event types names: %s\n", concat_set_in_order(event_names).c_str());
+		auto events_sc_names = libsinsp::events::sc_set_to_sc_names(events_sc_codes);
+		printf("-- Filter AST ppm sc names: %s\n", concat_set_in_order(events_sc_names).c_str());
 	}
```
Comment (Andreagit97) [inline userspace/libsinsp/examples/test.cpp:361]: why don't we move this directly inside `open_engine`? in this way, we could hide this ppm_sc patch

Comment (incertum) [reply userspace/libsinsp/examples/test.cpp:361]: We set the filter string right before and given this is an example and test binary, leaving it there should increase technical clarity. Unsure if hiding is what we want here.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**